### PR TITLE
Add config for Github Workflows

### PR DESCRIPTION
**Pull request by steelbrain**
⚠️ Currently targets #45 as base branch

Adds config to lint/test the main package and build/store the example repo. The built example zips can be downloaded by public from each commit's CI log

Fixes https://github.com/rhysmorgan134/node-CarPlay/issues/46

### CODE REVIEW

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+          restore-keys: |
+            ${{ runner.os }}-npm-
+
+      - name: Packager Version
+        run: npm version
+
+      - name: Install Dependencies
+        run: (npm install) && (cd examples/carplay-web-app && npm install)
+
+      - name: Lint sources
```
Comment (gozmanyoni) [inline .github/workflows/build_and_test.yml:36]: We should have a step to run typecheck as well

File: .github/workflows/build_and_test.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
+name: build_and_test
+
+on:
+  push:
+    branches:
+      - main
+      - master # TODO: Remove this in favor of main eventually?
+      - next
+  pull_request:
```
Comment (gozmanyoni) [inline .github/workflows/build_and_test.yml:9]: Would be good if we can run lint, typecheck and test on pull requests - to know things are going to break before merging (and later we can also disallow merging if the build fails)

Comment (steelbrain) [reply .github/workflows/build_and_test.yml:9]: It should run on pull requests once merged, or maybe you need to enable actions in your repository settings

**Comment by steelbrain** [comment]
@gozmanyoni I've updated the config to include Typechecking. Re PRs, it's already enabled to run the full config on PRs but it's not "activated" yet as I understand it.

It may get activated automatically once this PR is merged, or if the repository settings are updated; there's a bunch of settings around running actions on branches from 3rd party contributors, not sure how safe it is to enable.

For testing this, in my fork, I didn't have to enable anything besides merging it to one of the branches (next in my case) and it worked well there

**Review by gozmanyoni — APPROVED** [review]

**Comment by gozmanyoni** [comment]
@steelbrain since i merged my branch to next - could you refresh this with the latest changes there?

**Comment by steelbrain** [comment]
@gozmanyoni It should be up to date now, thanks!

**Comment by steelbrain** [comment]
@gozmanyoni Seems to be running! https://github.com/rhysmorgan134/node-CarPlay/actions/runs/6216538687

**Pull request merged**